src/proto/client/ent.rs

```diff
@@ -1,20 +1,25 @@
 use super::super::batch::{CommitBatch, GetBatchStatus, OpenBatch};
-use super::super::{single, BatchStatus, JobId, Progress, ProgressUpdate, Track};
+use super::super::{single, BatchStatus, JobId, Progress, ProgressUpdate};
 use super::{Client, ReadToken};
 use crate::ent::{Batch, BatchHandle, BatchId};
 use crate::error::{self, Error};
-use tokio::io::{AsyncBufRead, AsyncWrite};
+use crate::proto::FetchProgress;
 
-impl<S: AsyncBufRead + AsyncWrite + Unpin + Send> Client<S> {
+impl Client {
     /// Send information on a job's execution progress to Faktory.
-    pub async fn set_progress(&mut self, upd: ProgressUpdate) -> Result<(), Error> {
-        let cmd = Track::Set(upd);
-        self.issue(&cmd).await?.read_ok().await
+    pub async fn set_progress<P>(&mut self, upd: P) -> Result<(), Error>
+    where
+        P: AsRef<ProgressUpdate> + Sync,
+    {
+        self.issue(&upd).await?.read_ok().await
```

Comment:
If you change this to
```suggestion
        self.issue(upd.as_ref()).await?.read_ok().await
```
I think you can get rid of the
```rust
impl<P> FaktoryCommand for P
where
    P: AsRef<ProgressUpdate> + Sync,
```
in favor of just a `impl FaktoryCommand for ProgressUpdate`

Reply:
Updated